presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -125,6 +127,25 @@ class AgendaViewModel @Inject constructor(
                     )
                 }
             }
+
+            AgendaScreenEvent.OnLogoutClicked -> {
+                logoutCurrentUser()
+            }
+
+            is AgendaScreenEvent.OnOpenLogoutDropDownMenu -> {
+                _agendaScreenState.update { agendaScreenState ->
+                    agendaScreenState.copy(
+                        shouldOpenLogoutDropDownMenu = agendaScreenEvent.shouldOpen
+                    )
+                }
+            }
+        }
+    }
+
+    private fun logoutCurrentUser() {
+        viewModelScope.launch {
+            logoutUseCase.execute()
+            preferenceRepository.deleteCurrentUser()
```

Comment:
Do we need to do anything with the database tables?

Reply:
Delete them

What if a 2nd user logs in? Then that user will be able to see the previous user´s tables